api/v1beta1/cluster_types.go

```diff
@@ -493,6 +493,10 @@ type ClusterSpec struct {
 	// Version defines the temporal version the cluster to be deployed.
 	// This version impacts the underlying persistence schemas versions.
 	Version string `json:"version"`
+	// JobTtlSecondsAfterFinished is amount of time to keep job pods after jobs are completed.
+	// This field is immutable.
+	//+kubebuilder:validation:Minimum=1
+	JobTtlSecondsAfterFinished int32 `json:"jobTtlSecondsAfterFinished"`
```

Comment:
As it's only for persistence jobs IMO we could move this to the `PersistenceSpec` what do you think ? 
Note that the API refactor PR has been merged, so you'll have to rebase your branch :)

Reply:
as discussed we decided to leave it to not confuse user